speech/src/main/java/com/example/speech/InfiniteStreamRecognize.java

```diff
@@ -63,19 +61,37 @@ public class InfiniteStreamRecognize {
   private static boolean newStream = true;
   private static double bridgingOffset = 0;
   private static boolean lastTranscriptWasFinal = false;
+  private static boolean stopRecognition = false;
   private static StreamController referenceToStreamController;
   private static ByteString tempByteString;
```

Comment:
All constants in Java should use [UPPER_SNAKE_CASE](https://engdoc.corp.google.com/eng/doc/devguide/java/style/index.md?cl=head#s5.2.4-constant-names)

Reply:
I don't have google acc :(

Reply:
@minherz By constant, do you mean the variable marked as final or static is constant too?

Reply:
> I don't have google acc :(

I looked into the case closer. It looks like you made changes to the code sample itself and not only added the test case. I will ask a Googler who opened the issue to review the changes you made to the code sample itself.


> @minherz By constant, do you mean the variable marked as final or static is constant too?

I mean "constants" in Java. It should be literals declared with `const` or objects defined with `final`. Static or not static should not play a role.

Reply:
Yeap, I refactored this sample to have a possibility to replace the voice recognition thread with a file or another source of an audio byte stream